File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        int counter=0;
        int ksmallest=INT_MIN;
        inorder(root,counter,k,ksmallest);
        return ksmallest;
    }
    void inorder(TreeNode* root,int &counter,int k,int &ksmallest){
        if(root==NULL|| counter>=k)return;
        inorder(root->left,counter,k,ksmallest);
        counter++;
        if(counter==k){
            ksmallest=root->val;
            return;
        }
        inorder(root->right,counter,k,ksmallest);
    }
};

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:

        stack = []
        current = root
        
        while True:
            # Go to the leftmost node
            while current:
                stack.append(current)
                current = current.left
            
            # Process node
            current = stack.pop()
            k -= 1
            
            # If kth element found
            if k == 0:
                return current.val
            
            # Move to right subtree
            current = current.right